shakenfist/db.py

```diff
@@ -146,13 +161,86 @@ def persist_ipmanager(network_uuid, data):
 def delete_ipmanager(network_uuid):
     etcd.delete('ipmanager', None, uuid)
 
+#####################################################################
+# Instances
+#####################################################################
+
+# Theorem: we never write to the instance object, except for when we
+# create or delete it. Anything which changes value during the lifetime
+# of the instance (state, power_state, etc) is stored in a set of
+# secondary values in etcd that we never cache. This is version 2 of
+# the instance object.
+
+
+def create_instance(instance_uuid, cpus, disk_spec, memory, name, namespace,
+                    requested_placement, ssh_key, user_data, video):
+    metadata = {
```

Comment:
I think it would be better to keep object metadata generation in the class code, like `Network.metadata()`

Reply:
I can do that. I think I might try and get tests passing at all first, but make sure I don't forget.

Reply:
This is done in the next version.